doc/developer/design/20230607_shift_status_updates.md

```diff
@@ -0,0 +1,151 @@
+
+Due to the fact that there is currently no way for these dataflows to gracefully shutdown, dropping their backing replicas does not result in status updates indicating that work has been paused.
+
+A concrete example of this can be witnessed by creating in a source in some cluster, dropping the only replica in that cluster, and inspecting it status via `mz_internal.mz_source_statuses`. The current behavior is that the source is incorrectly described as `running` even though it is not possible for ingestion to occur.
+
+## Goals
+
+<!--
+Enumerate the concrete goals that are in scope for the project.
+-->
+
+- Describe how to shift the writing of status updates for sources/sinks from storage workers to the storage controller
+- How to evolve status relations to support storage clusters with multiple replicas
+
+<!-- ## Non-Goals -->
+
+<!--
+Enumerate potential goals that are explicitly _out_ of scope for the project
+ie. what could we do or what do we want to do in the future - but are not doing
+now.
+-->
+
+## Overview
+
+<!--
+Brief, high-level overview. A few sentences long, at most a couple of smaller
+paragraphs.
+-->
+
+The primary motivation for enforcing status updates flow through the storage controller is that it allows us to correctly handle instances in which a replica is dropped, in which case any late status updates can simply be ignored and the status of any dataflows managed by the dropped replica can have their status set to `paused`. As mentioned in the [Alternatives](#alternatives) section, relying on graceful shutdown instead would be complex and error prone.
+
+The health operator for ingestion dataflows will continue to exist as a terminal operator in the dataflow graph. However, rather than writing status updates directly to the persist shard corresponding to the collection's status history, the health operator will update a local mapping from `GlobalId` to `HealthStatus` that exists as part of the storage worker's state. Updates of a similar fashion will be done during sink production as well.
+
+Upon some defined frequency, storage workers will emit a new `StorageResponse::StatusUpdates` to the storage controller. When receiving these events, the storage controller will write the status updates to the shard associated with `mz_source_status_history` / `mz_sink_status_history`.
+
+## Detailed description
+
+<!--
+Describe the approach in detail. If there is no clear frontrunner, feel free to
+list all approaches in alternatives. If applicable, be sure to call out any new
+testing/validation that will be required.
+
+For some features it can be helpful to sketch an implementation. If you're
+working on things that are crossing team boundaries it will be helpful to spell
+out any new interfaces/traits/interactions.
+
+For most new features, you should think about testing, rollout/lifecycle, and
+observability. These things can warrant their own sections.
```

Comment:
It's probably worth expanding on testing here a bit. We have some existing source/sink status tests; these ought to carry over pretty well, though we'll at least want to add tests for the new `paused` behaviour. I'm not sure if there's any other testing we can fruitfully do here...